src/Workspaces/CSharp/Portable/CodeGeneration/ArgumentGenerator.cs

```diff
@@ -20,12 +23,12 @@ public static ArgumentSyntax GenerateArgument(SyntaxNode argument)
 
         public static ArgumentListSyntax GenerateArgumentList(IList<SyntaxNode> arguments)
         {
-            return SyntaxFactory.ArgumentList(SyntaxFactory.SeparatedList(arguments.Select(GenerateArgument)));
+            return SyntaxFactory.ArgumentList(SyntaxFactory.SeparatedList(arguments.Select(s_generateArgument)));
```

Comment:
Are you seeing these allocations in any traces?  I'm not a fan of cluttering the code in this fashion if it's not in a hotpath.